server/service/integration_mdm_test.go

```diff
@@ -5316,7 +5316,178 @@ func (s *integrationMDMTestSuite) TestGetPoliciesRequestWithNotElegibleHost() {
 	err = xml.Unmarshal(resBytes, &xmlType)
 	require.NoError(t, err)
 
-	// Checking if SOAP response contains a valid DiscoveryResponse message
+	// Checking if SOAP response contains a valid SoapFault message
+	resSoapMsg := string(resBytes)
+	require.True(t, s.isXMLTagPresent("s:fault", resSoapMsg))
+	require.True(t, s.isXMLTagContentPresent("s:value", resSoapMsg))
+	require.True(t, s.isXMLTagContentPresent("s:text", resSoapMsg))
```

Comment:
We're starting to do a number of failed calls tests where we only check for the presence of an `s:fault` tag with some content, would it make sense to start asserting that the error text is the expected one (we typically do it with just checking that the error message contains a specific substring, not necessarily a full string equality)? In case it fails but due to unrelated conditions vs what we setup to test.

Reply:
This is a great point and I agree that checking the actual error is a good thing. I've just added the code to do this on all the tests that generates a SoapFault